rskj-core/src/fuzzTest/java/co/rsk/net/discovery/PacketDecoderFuzzTest.java

```diff
@@ -0,0 +1,121 @@
+
+    @Tag("DiscoveryPacketDecoderFuzzDecode")
+    @FuzzTest
+    public void testFuzzDecode(FuzzedDataProvider data) {
+        ECKey key = ECKey.fromPrivate(Hex.decode(KEY_1)).decompress();
+        String check = UUID.randomUUID().toString();
+        ChannelHandlerContext ctx = Mockito.mock(ChannelHandlerContext.class);
+
+        PacketDecoder decoder = new PacketDecoder();
+
+        PingPeerMessage pingMessage = PingPeerMessage.create(data.consumeString(256), data.consumeInt(), check, key, data.consumeInt());
+        InetSocketAddress sender = new InetSocketAddress(data.consumeString(256), data.consumeInt(0, 65535));
+        long start = System.currentTimeMillis();
+        try {
+            decoder.decodeMessage(ctx, pingMessage.getPacket(), sender);
+        } catch (Exception e) {}
+
+        //Decode Pong Message
+        PongPeerMessage pongPeerMessage = PongPeerMessage.create(data.consumeString(256), data.consumeInt(), check, key, data.consumeInt());
+        sender = new InetSocketAddress(data.consumeString(256), data.consumeInt(0, 65535));
+        try {
+            decoder.decodeMessage(ctx, pongPeerMessage.getPacket(), sender);
+        } catch (Exception e) {}
+
+        //Decode Find Node Message
+        FindNodePeerMessage findNodePeerMessage = FindNodePeerMessage.create(key.getNodeId(), check, key, data.consumeInt());
+        sender = new InetSocketAddress(data.consumeString(256), data.consumeInt(0, 65535));
+        try {
+            decoder.decodeMessage(ctx, findNodePeerMessage.getPacket(), sender);
+        } catch (Exception e) {}
+
+        //Decode Neighbors Message
+        NeighborsPeerMessage neighborsPeerMessage = NeighborsPeerMessage.create(new ArrayList<>(), check, key, data.consumeInt());
+        sender = new InetSocketAddress(data.consumeString(256), data.consumeInt(0, 65535));
+        try {
+            decoder.decodeMessage(ctx, neighborsPeerMessage.getPacket(), sender);
+        } catch (Exception e) {}
+        long end = System.currentTimeMillis();
+
+        if (end - start > 250) {
+            System.out.println("It took too much");
+        }
+    }
+
+    @Tag("DiscoveryPacketDecoderFuzzDecode2")
+    @FuzzTest
+    public void testFuzzDecode2(FuzzedDataProvider data) {
+        ChannelHandlerContext ctx = Mockito.mock(ChannelHandlerContext.class);
+        InetSocketAddress sender = new InetSocketAddress(data.consumeString(256), data.consumeInt(0, 65535));
+        PacketDecoder decoder = new PacketDecoder();
+
+        long start = System.currentTimeMillis();
+        byte[] fuzzData = data.consumeBytes(4096);
+        byte[] mdcCheck = keccak256(fuzzData, 0, fuzzData.length);
+        byte[] msg = Arrays.concatenate(mdcCheck, fuzzData);
+        try {
+            decoder.decodeMessage(ctx, msg, sender);
+        } catch (Exception e) {
+            // Messages will fail really most of the times.
```

Comment:
Why is this valid? I think this should be implemented now, documenting in the code why it is done like that.